micro-rdk/src/common/conn/viam.rs

```diff
@@ -837,8 +843,18 @@ where
     ) -> Task<Result<(), errors::ServerError>> {
         let exec = self.executor.clone();
         let robot = self.robot.clone();
+
+        // If the connection manager has a low limit on the number of
+        // concurrent connections, don't enable local signaling. This
+        // value may need to be adjusted if the logic in
+        // `ViamServerBuilder::get_default_max_concurrent_connections`
+        // changes.
         #[cfg(feature = "local-signaling")]
-        let ss = self.local_signaling_server.clone();
+        let ss = self
+            .local_signaling_server
+            .clone()
+            .filter(|_| self.incomming_connection_manager.max_connections() >= 5);
```

Comment:
that's pretty cool, forgot options can use `filter`. this way the `max_connections()` check only happens if `self.local_signaling_server.is_some()`.

My instinct would have been to go with this
```rs
let ss = if self.incomming_connection_manager.max_connections() >= 5 { 
    self.local_signaling_server.clone() 
} else {
    None
}
``` 

Did you still want to add the warning though?